webapp/src/main/java/com/box/l10n/mojito/service/smartling/ThirdPartyTextUnitMatchingService.java

```diff
@@ -0,0 +1,173 @@
+package com.box.l10n.mojito.service.smartling;
+
+import com.box.l10n.mojito.entity.TMTextUnit;
+import com.box.l10n.mojito.entity.ThirdPartyTextUnit;
+import com.box.l10n.mojito.service.tm.TMTextUnitRepository;
+import com.box.l10n.mojito.smartling.SmartlingClient;
+import com.box.l10n.mojito.smartling.request.File;
+import com.box.l10n.mojito.smartling.request.StringData;
+import com.box.l10n.mojito.smartling.response.FilesResponse;
+import com.box.l10n.mojito.smartling.response.SourceStringsResponse;
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Service to manage matching between mojito text units and third party text units
+ **/
+@Service
+public class ThirdPartyTextUnitMatchingService {
+    /**
+     * logger
+     */
+    private static Logger logger = getLogger(ThirdPartyTextUnitMatchingService.class);
+
+    @Autowired
+    ThirdPartyTextUnitRepository thirdPartyTextUnitRepository;
+
+    @Autowired
+    SmartlingClient smartlingClient;
+
+    @Autowired
+    TMTextUnitRepository tmTextUnitRepository;
+
+    void getSmartlingThirdPartyTextUnits(String projectId) {
+        FilesResponse files = smartlingClient.getFiles(projectId);
+        for (File file : files.getResponse().getData().getItems()) {
+            String fileUri = file.getFileUri();
+            if (fileUri.matches(".*\\d+_(singular|plural)_source\\.xml")) {
```

Comment:
Can you please move the regex to a pattern that is created only once and replace this with something like `pattern.match(fileUri).matches()`